External/Plugins/ASCompletion/Model/ASFileParser.cs

```diff
@@ -1705,8 +1705,18 @@ private bool LookupRegex(string ba, ref int i)
             while (i0 > 0)
             {
                 c = ba[i0--];
-                if ("=(,[{;?:+-*/|&~^><n!".IndexOf(c) >= 0) break; // ok
-                if (" \t".IndexOf(c) >= 0) continue;
+                if (c == '=' || c == '(' || c == '[' || c == '{' || c == ',' || c == ';'
+                    || c == '?' || c == ':'
+                    || c == '+' || c == '-' || c == '*' || c == '/'
+                    || c == '|' || c == '&' || c == '~' || c == '^'
+                    || c == '>' || c == '<' || c == '!'
+                    || c == 'n' /* return ~/regex/ or in ~/regex/ */
+                    || c == 'h' /* switch ~/regex/ */
+                    || c == 'y' /* try ~/regex/ */)
```

Comment:
I think this is way less readable than previous code, but it's not so big deal. The problem I see now is that this is not quite bulletproof (plus that single character check makes me feel a bit uneasy heh). For example, what if:

`function test():Void ~/regex/;`

In the same way it would be possible to do for example:

`function test(value:String):AnythingPossible ~/regex/.match(value) ? return whatever : return null;`

Reply:
Yes, such cases are not currently supported.
The parsing of single-line functions I will be make separately.